test/integration/translation_failures_test.go

```diff
@@ -128,3 +213,57 @@ func pluginUsingInvalidCACert(ns string) *kongv1.KongPlugin {
 		PluginName: "mtls-auth",
 	}
 }
+
+func httpRouteWithBackends(gatewayName string, services ...*corev1.Service) *gatewayv1beta1.HTTPRoute {
+	httpPort := gatewayv1beta1.PortNumber(80)
+	weight := int32(100 / len(services))
+	pathMatchPrefix := gatewayv1beta1.PathMatchPathPrefix
+
+	backendRefs := make([]gatewayv1beta1.HTTPBackendRef, 0, len(services))
+	for _, service := range services {
+		backendRefs = append(backendRefs,
+			gatewayv1beta1.HTTPBackendRef{
+				BackendRef: gatewayv1beta1.BackendRef{
+					BackendObjectReference: gatewayv1beta1.BackendObjectReference{
+						Name: gatewayv1beta1.ObjectName(service.Name),
+						Port: &httpPort,
+						Kind: util.StringToGatewayAPIKindPtr("Service"),
+					},
+					Weight: &weight,
+				},
+			})
+	}
+
+	return &gatewayv1beta1.HTTPRoute{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: randomName(),
+			Annotations: map[string]string{
+				annotations.AnnotationPrefix + annotations.StripPathKey: "true",
+			},
+		},
+		Spec: gatewayv1beta1.HTTPRouteSpec{
+			CommonRouteSpec: gatewayv1beta1.CommonRouteSpec{
+				ParentRefs: []gatewayv1beta1.ParentReference{{
+					Name: gatewayv1beta1.ObjectName(gatewayName),
+				}},
+			},
+			Rules: []gatewayv1beta1.HTTPRouteRule{
+				{
+					Matches: []gatewayv1beta1.HTTPRouteMatch{
+						{
+							Path: &gatewayv1beta1.HTTPPathMatch{
+								Type:  &pathMatchPrefix,
+								Value: kong.String("/test"),
+							},
+						},
+					},
+					BackendRefs: backendRefs,
+				},
+			},
+		},
+	}
+}
+
+func randomName() string {
```

Comment:
what about getting the prefix as an argument and moving this helper to a common util package?

Reply:
f56018c